datashape/coretypes.py

```diff
@@ -402,6 +402,70 @@ def to_numpy_dtype(self):
         return np.dtype('O', metadata={'vlen': unicode})
 
 
+class Decimal(Unit):
+    """Decimal type corresponding to SQL Decimal/Numeric types.
+
+    The first parameter passed specifies the number of digits of precision that
+    the Decimal contains. If an additional parameter is given, it represents
+    the scale, or number of digits of precision that are after the decimal
+    point.
+
+    The Decimal type makes no requirement of how it is to be stored in memory,
+    therefore, the number of bytes needed to store a Decimal for a given
+    precision will vary based on the platform where it is used.
+
+    >>> Decimal(18)
+    decimal[18,0]
+    >>> Decimal(7,4)
+    decimal[7,4]
+    """
+
+    cls = MEASURE
+    __slots__ = 'precision', 'scale'
+
+    def __init__(self, precision, *args):
+        if len(args) == 0:
+            scale = 0
+        elif len(args) == 1:
+            scale = args[0]
+        else:
+            raise TypeError('Too many arguments passed to Decimal')
+
+        self.precision = precision
+        self.scale = scale
+
+    def __str__(self):
+        return 'decimal[{0},{1}]'.format(self.precision, self.scale)
+
+    def __repr__(self):
+        return str(self)
+
+    def to_numpy_dtype(self):
+        """
+        Note that floating-point (scale > 0) precision will be lost converting
```

Comment:
So what I mean by this is that `123.45` in a `Decimal(5,2)` is an _exact_ representation of 123.45. When it gets converted to a `float64`, technically it will be some binary representation that is incredibly close to but not necessarily exactly 123.45, like `123.450000000002312908`. Maybe that's obvious, but I wanted to point out in the function that that happens.

I don't see this as an "error" per se, just a natural function of representing floating point numbers in an environment without decimal types. I would still be able to convert from data that had `scale > 0` decimals to NumPy-compatible types.